### PR TITLE
feat: add `php please` spec for Statamic 3

### PR DESCRIPTION
**Pull request by martyf**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Spec addition: Statamic 3's `please` command

**What is the current behavior? (You can also link to an open issue here)**
Currently `please` is not available.

**What is the new behavior (if this is a feature change)?**
Add's Statamic 3's `please` spec based on the command list json output

**Additional info:**
This is replicating the setup of the `artisan` spec for Laravel - perfect fit as Statamic is built on top of Laravel. Full credit to the original authors of the `artisan` implementation for this - this PR operates the same way, but runs the `please` command instead. This includes:
- only adding the spec if `please` is in the current directory, 
- pulls the available options from the `php please list --format=json` command

### CODE REVIEW

File: src/php/please.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const completionSpec: Fig.Spec = {
+  name: "please",
+  description: "Statamic Please command",
+  generateSpec: async (tokens, executeShellCommand) => {
+    var out = await executeShellCommand("php please list --format=json");
```
Comment (QuiiBz) [inline src/php/please.ts:5]: Could you replace this with a `const`?
```suggestion
    const out = await executeShellCommand("php please list --format=json");
```

File: src/php/please.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+            return {
+              name: names,
+              description: option.description,
+            };
+          }),
+        });
+      });
+    } catch (err) {
+      //
```
Comment (QuiiBz) [inline src/php/please.ts:41]: We should add a `console.error(err);` in the catch, to be able to debug the spec in the future

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks for this first PR! We have only two small issues to resolve before merging this 🚀

**Comment by martyf** [comment]
Thanks @QuiiBz - I've just updated the PR with these two tweaks.

File: src/php/please.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    try {
+      const commandDefinition = JSON.parse(out);
+
+      commandDefinition.commands.map((command) => {
+        subcommands.push({
+          name: command.name,
+          description: command.description,
+
+          args: Object.keys(command.definition.arguments).map((argumentKey) => {
```
Comment (fedeci) [inline src/php/please.ts:16]: nit: we can directly use `Object.values` to simplify this logic

File: src/php/please.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            const argument = command.definition.arguments[argumentKey];
+
+            return {
+              name: argument.name,
+              description: argument.description,
+              isOptional: !argument.is_required,
+            };
+          }),
+          options: Object.keys(command.definition.options).map((optionKey) => {
```
Comment (fedeci) [inline src/php/please.ts:25]: Same as above

**Comment by martyf** [comment]
Tweaked to use `Object.values` instead of `Object.keys`

**Review by fedeci — APPROVED** [review]

**Review by QuiiBz — APPROVED** [review]
This looks amazing, thank you so much for this contribution 🎉

**Pull request merged**